### PR TITLE
07 post api articles article id comments

### PR DESCRIPTION
**Pull request by lemonses**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -122,4 +123,54 @@ describe('GET /api/articles',()=>{
...
+        .then(({body}) => {
+            const comment = body.comment
+            expect(typeof comment.comment_id).toBe('number')
+            expect(typeof comment.votes).toBe('number')
+            expect(typeof comment.created_at).toBe('string')
+            expect(typeof comment.author).toBe('string')
+            expect(typeof comment.body).toBe('string')
+            expect(typeof comment.article_id).toBe('number')
+        })
+    })
```
Comment (B7Bprog) [inline __tests__/server.test.js:144]: Please add another test where extra properties of the post object are ignored.

For example when posting with this:
`{username:'rogersop',body:'life changing',extraProperty:'extraValue'}`

File: __tests__/server.test.js [hunk 2]
```diff
@@ -122,4 +123,54 @@ describe('GET /api/articles',()=>{
...
+    test('should return a 400 bad request if passed a username that does not exist in the users database',()=>{
```
Comment (B7Bprog) [inline __tests__/server.test.js]: This should be a 404, as the user can be added later but currently doesn't exist. Please change this test and code.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Wilf, nice work so far, only a couple of changes to make, please see my comments. Once it's all done and all the tests passing, please send this PR back to us with the update.

File: __tests__/server.test.js (hunk 1 above)
Comment (lemonses) [reply __tests__/server.test.js:144]: I knew I was missing a test somewhere thanks!

File: __tests__/server.test.js [hunk 3]
```diff
@@ -173,4 +174,21 @@ describe('POST /api/articles/:article_id/comments',()=>{
...
+            const comment = body.comment
+            expect(comment).toMatchObject({
+                comment_id : 21,
+                votes : 0,
+                created_at: convertTimestampToDate(Date.now()),
+                author : 'rogersop',
+                body : 'life changing',
+                article_id: 11
+            })
+        })
```
Comment (Jodiech) [inline __tests__/server.test.js:240]: Nice work on amending this test 👍

File: __tests__/server.test.js [hunk 4]
```diff
@@ -141,12 +142,12 @@ describe('POST /api/articles/:article_id/comments',()=>{
...
         })
     })
-    test('should return a 400 bad request if passed a username that does not exist in the users database',()=>{
+    test('should return a 404  User not found if passed a username that does not exist in the users database',()=>{
         return request(app)
         .post('/api/articles/12/comments')
         .send({username:'notAUser',body:'life changing'})
-        .expect(400).then(({body})=>{
-            expect(body.message).toBe('Bad request')
+        .expect(404).then(({body})=>{
```
Comment (Jodiech) [inline __tests__/server.test.js:149]: Nice 🤸

**Review by Jodiech — APPROVED** [review]
Go ahead and merge! Well done! 🚀

**Pull request merged**